lib/pdk/control_repo.rb

```diff
@@ -0,0 +1,45 @@
+require 'pdk'
+
+module PDK
+  module ControlRepo
+    CONTROL_REPO_FILES = %w[Puppetfile environment.conf].freeze
+
+    # Returns path to the root of the Control Repo being worked on.
+    #
+    # An environment.conf is required for a Control Repo, whereas Puppetfile is
+    # optional. Note that a Bolt Project can also be a Control Repo.
+    #
+    # @see https://puppet.com/docs/pe/latest/control_repo.html
+    #
+    # @param strict_check [Boolean] When strict_check is true, only return the path
+    #   if the Control Repo is strictly _only_ a control repository. For example,
+    #   not also a Puppet Bolt project directory  Default is false.
+    #
+    # @return [String, nil] Fully qualified base path to Control Repo, or nil if
+    #   the current working dir does not appear to be within a Control Repo.
+    def control_repo_root(strict_check = false)
+      environment_conf_path = PDK::Util.find_upwards('environment.conf')
+      path = if environment_conf_path
+               File.dirname(environment_conf_path)
+             elsif in_control_repo_root?(Dir.pwd)
+               Dir.pwd
+             else
+               nil
+             end
+      return path if path.nil? || !strict_check
+      PDK::Bolt.in_bolt_project_root?(path) ? nil : path
+    end
+    module_function :control_repo_root
+
+    # Returns true or false depending on if any of the common files in a Control Repo
+    # are found in the specified directory. If a directory is not specified, the current
+    # working directory is used.
+    #
+    # @return [boolean] True if any folders from MODULE_FOLDERS are found in the current dir,
+    #   false otherwise.
+    def in_control_repo_root?(path = Dir.pwd)
```

Comment:
```suggestion
    def is_control_repo_root?(path = Dir.pwd)
```

Same as in the Bolt code, I think "is" makes slightly more sense given you can pass any arbitrary path.

Reply:
Changed to `control_repo_root?`